src/main/java/net/openhft/hashing/LongTupleHashFunction.java

```diff
@@ -0,0 +1,852 @@
+ *
+ * The {@link #bitsLength} method should also be implemented, returning the actual number of bits in
+ * the result array. The bits length should be greater than 64, otherwise just use the
+ * {@link LongHashFunction} interface. And the length should also be a positive multiple of 8.
+ *
+ * @see LongHashFunction
+ * for additional information about subclassing andd access.
+ */
+@ParametersAreNonnullByDefault
+public abstract class LongTupleHashFunction implements Serializable {
+    private static final long serialVersionUID = 0L;
+
+    // Implementations
+    //
+
+    /**
+     * Returns a 128-bit hash function implementing
+     * <a href="https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp">MurmurHash3
+     * algorithm</a> without seed values. This implementation produces equal results for equal input
+     * on platforms with different {@link ByteOrder}, but is slower on big-endian platforms than on
+     * little-endian.
+     *
+     * @see #murmur_3(long)
+     */
+    @NotNull
+    public static LongTupleHashFunction murmur_3() {
+        return MurmurHash_3.asLongTupleHashFunctionWithoutSeed();
+    }
+
+    /**
+     * Returns a 128-bit hash function implementing
+     * <a href="https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp">MurmurHash3
+     * algorithm</a> with the given seed value. This implementation produces equal results for equal
+     * input on platforms with different {@link ByteOrder}, but is slower on big-endian platforms
+     * than on little-endian.
+     *
+     * @see #murmur_3()
+     */
+    @NotNull
+    public static LongTupleHashFunction murmur_3(final long seed) {
+        return MurmurHash_3.asLongTupleHashFunctionWithSeed(seed);
+    }
+
+    /**
+     * Constructor for use in subclasses.
+     */
+    protected LongTupleHashFunction() {}
+
+    // Public API
+    //
+
+    /**
+     * Returns the actual number of bits in a result array; a positive multiple of 8.
+     */
+    public abstract int bitsLength();
+
+    /**
+     * Returns a new-allocated result array.
+     */
+    @NotNull
+    public long[] newResultArray() {
+        return new long[(bitsLength() + 63) / 64];
+    }
+
+    /**
+     * Computes the hash code for the given {@code long} value, and store the results in the
+     * {@code result} array; this method is consistent with {@code LongTupleHashFunction} methods
+     * that accept sequences of bytes, assuming the {@code input} value is interpreted in
+     * {@linkplain ByteOrder#nativeOrder() native} byte order. For example, the result of
+     * {@code hashLong(v, result)} call is identical to the result of
+     * {@code hashLongs(new long[] {v}, result)} call for any {@code long} value.
+     *
+     * The {@code result} array should be always created by {@link #newResultArray} method.
+     * When storing, the {@code result[0 .. newResultArray().length-1]} will be accessed,
```

Comment:
These docs don't specify what happens when the result length is not a multiple of 8 bytes, say 10 bytes. This is perfectly fine, but then the class-level javadoc and the javadoc for `bitsLength` should mention "a multiple of 64" instead of "multiple of 8", and also the round-up in `newResultArray()` implementation is not needed

Reply:
according to [wiki](https://en.wikipedia.org/wiki/List_of_hash_functions) there are some hash with variable length or non 64-multiple bit length, e.g. sha-1, sha-224. So a multiple of 8 could be flexible.